### PR TITLE
QA-180 Parent POM + Managed Profile

### PR DESCRIPTION
**Pull request by MattGill98**
From most notable to least:

- Added a managed profile (as the default).
- Added a parent pom to run all TCKs against a profile, and refactored profiles accordingly.
- Removed Payara 4 related profiles and folders. A new branch would need adding where a backport of this fix can be made to only target Payara 4.
- Moved all tests from the `test` phase to the `integration-test` phase.
- Changed most of the modules to from JAR packaging to pom.
- Standardised project artifact names.
- Minor changes to `.gitignore`.

In short, all TCKs should now be runnable against a Payara Server version with the following command:

~~~
mvn clean verify -Dpayara.version=5.x
~~~

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+        <module>MicroProfile-Health/</module>
+        <module>MicroProfile-Config/tck-runner</module>
+        <module>MicroProfile-OpenAPI/tck-runner</module>
+    </modules>
+
+    <dependencies>
+        <!-- Arquillian Dependencies -->
+        <dependency>
+            <groupId>org.jboss.arquillian.container</groupId>
+            <artifactId>arquillian-container-test-spi</artifactId>
```
Comment (MarkWareham) [inline pom.xml]: Where are the versions for the dependencies? - Perhaps a dependency management section is needed.

File: MicroProfile-Fault-Tolerance/tck-runner/pom.xml [hunk 2]
```diff
@@ -96,19 +37,37 @@
...
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
 
+    <profiles>
         <profile>
             <id>stable</id>
-
+            <activation>
```
Comment (MarkWareham) [inline MicroProfile-Fault-Tolerance/tck-runner/pom.xml]: This is only for Jenkins' benefit. This should not be activated by default.

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,314 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    
+    <groupId>fish.payara.microprofile</groupId>
+    <artifactId>tck-suite-parent</artifactId>
+    <version>1.0-SNAPSHOT</version>
```
Comment (MarkWareham) [inline pom.xml]: I would suggest increasing this. The old individual poms were this version.

File: MicroProfile-JWT-Auth/tck-arquillian-extension/pom.xml [hunk 4]
```diff
@@ -0,0 +1,83 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS HEADER.
+
+Copyright (c) 2017 Payara Foundation and/or its affiliates. All rights reserved.
```
Comment (MarkWareham) [inline MicroProfile-JWT-Auth/tck-arquillian-extension/pom.xml]: years

File: pom.xml (hunk 1 above)
Comment (MattGill98) [reply pom.xml]: They're specified by the arquillian bom further up!

File: MicroProfile-JWT-Auth/tck-arquillian-extension/pom.xml (hunk 4 above)
Comment (MattGill98) [reply MicroProfile-JWT-Auth/tck-arquillian-extension/pom.xml]: Good shout, I need to update all of the copyright headers.

File: MicroProfile-Fault-Tolerance/tck-runner/pom.xml (hunk 2 above)
Comment (MattGill98) [reply MicroProfile-Fault-Tolerance/tck-runner/pom.xml]: Similar to discussion on https://github.com/payara/patched-src-microprofile1.2-samples/pull/7#discussion_r243356674. I'm in favour of defaulting to stable profiles. What are your thoughts?

Comment (MarkWareham) [reply MicroProfile-Fault-Tolerance/tck-runner/pom.xml]: I agree projects should be stable by default. Though running a project with -P \!stable isn't very intuitive. My preference is for a "also-run-unstable-tests"

**Comment by MarkWareham** [comment]
Shall we merge this? - I believe it's a breaking change (directory paths & maven stage changes)

**Comment by MattGill98** [comment]
Should now support Payara 4 again (but without the need for the profile activation).

Depends on https://github.com/payara/Payara_PatchedProjects/pull/214.

@MarkWareham

File: MicroProfile-Rest-Client/tck-runner/pom.xml [hunk 5]
```diff
@@ -91,6 +77,48 @@
     </build>
 
     <profiles>
+        <profile>
+            <id>payara5</id>
+            <activation>
+                <property>
+                    <name>payara.version</name>
```
[the remark this answers is not in the document]
Comment (MattGill98) [reply MicroProfile-Rest-Client/tck-runner/pom.xml]: That's not true, it works on project properties and can be overwritten by command line properties as one might expect

File: MicroProfile-OpenAPI/tck-runner/pom.xml [hunk 6]
```diff
@@ -60,114 +86,75 @@
...
+                            </environmentVariables>
                         </configuration>
                     </plugin>
                 </plugins>
             </build>
         </profile>
 
+        <profile>
+            <id>payara-server-remote</id>
+        </profile>
```
[the remark this answers is not in the document]
Comment (MattGill98) [reply MicroProfile-OpenAPI/tck-runner/pom.xml]: I don't believe so, in fact that empty block can be removed!

**Comment by MattGill98** [comment]
Working out if it's possible to pull the extension from patched projects. If not then this will depend on a maven central upload.

**Comment by MarkWareham** [comment]
needs to coincide with Jenkins pipeline change. There is a PR for that too.
